shm/source/Map.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Map.hpp"
+
+
+Map::Map():
+	horizonLimit_(100),
+	islands_(generateIslands(10)),
+	currentPosition_()
+{}
+
+Island* Map::getIsland(const Island::Coordinates& coordinate) {
+	auto result = std::find_if(islands_.begin(), islands_.end(), [coordinate](const Island& island){ return island.getPosition() == coordinate; });
+	if (result != islands_.end()) {
+		return &(*result);
+	}
+	return nullptr;
+}
+
+std::vector<Island> Map::generateIslands(const size_t numberOfIslands) const {
+	if (numberOfIslands < 1) {
+		return {};
+	}
+	auto xyPositions = generateRandomPositions(numberOfIslands * 2);		
+	std::vector<Island> islands;
+	islands.reserve(xyPositions.size() / 2);
+	for (size_t i = 1; i < xyPositions.size(); i += 2) {
+		islands.emplace_back(Island::Coordinates(xyPositions.at(i - 1), xyPositions.at(i)));
```

Comment:
when you use ecmplae_back you provide arguments immediately to C'tor, if you want to create an object and insert it better use push_back.

like vec.empalce_back(5, "ala");

Reply:
In other cases, you use copy or move C'tor...